src/main/kotlin/com/expedia/graphql/schema/extensions/annotationExtensions.kt

```diff
@@ -79,21 +77,24 @@ internal fun KAnnotatedElement.directives(hooks: SchemaGeneratorHooks) =
 
 @Throws(CouldNotGetNameOfAnnotationException::class)
 private fun DirectiveInfo.getGraphQLDirective(hooks: SchemaGeneratorHooks): GraphQLDirective {
-    val kClass: KClass<out DirectiveAnnotation> = this.annotation.annotationClass
-    val builder = GraphQLDirective.newDirective()
-    val name: String = this.effectiveName ?: throw CouldNotGetNameOfAnnotationException(kClass)
+    val directiveClass = this.directive.annotationClass
+    val name: String = this.effectiveName ?: throw CouldNotGetNameOfAnnotationException(directiveClass)
 
     @Suppress("Detekt.SpreadOperator")
+    val builder = GraphQLDirective.newDirective()
+        .name(name.normalizeDirectiveName())
+        .validLocations(*this.directiveAnnotation.locations)
+        .description(this.directiveAnnotation.description)
 
-    builder.name(name.normalizeDirectiveName())
-        .validLocations(*this.annotation.locations)
-        .description(this.annotation.description)
+    directiveClass.getValidProperties(hooks).forEach { property ->
+        val propertyName = property.name
+        val value = property.call(this.directive)
 
-    kClass.getValidFunctions(hooks).forEach { kFunction ->
-        val propertyName = kFunction.name
-        val value = kFunction.call(kClass)
         @Suppress("Detekt.UnsafeCast")
-        val type = defaultGraphQLScalars(kFunction.returnType) as GraphQLInputType
+        var type: GraphQLInputType? = defaultGraphQLScalars(property.returnType) as? GraphQLInputType
```

Comment:
can you rewrite this as (and remove the suppression)
```
val type = defaultGraphQLScalars(prop.returnType) ?: hooks.willGenerateGraphQLType(prop.returnType)
val argument = GraphQLArgument.newArgument()
    .name(propertyName)
    .value(value)
    .type(type as? GraphQLInputType)
    .build()
```

Reply:
Oh that's much nicer :) :+1: 

ba1c4f1080082643d5c179d0d21ef3c4a1356eb6